### PR TITLE
fix repeate set chassis

### LINKED ISSUE #3084: fix repeat set sb db
# Expected Behavior

# Actual Behavior

sometime the chassis is already set, but sb db may not be writeable

``` bash 


I0727 15:35:32.055555       1 init.go:499] take 402.49 seconds to initialize IPAM
W0727 15:35:34.048587       1 ovn-sbctl.go:49] ovn-sbctl command took too long: ovn-sbctl --timeout=60 --db=tcp:[10.251.142.69]:6642,tcp:[10.251.142.152]:6642,tcp:[10.251.142.153]:6642 --format=csv --no-heading --data=bare --columns=name find chassis name=a8a69faf-03e1-4762-89b3-f8cf6c29fe24 in 1992ms
W0727 15:35:44.121648       1 ovn-sbctl.go:49] ovn-sbctl command took too long: ovn-sbctl --timeout=60 --db=tcp:[10.251.142.69]:6642,tcp:[10.251.142.152]:6642,tcp:[10.251.142.153]:6642 set chassis a8a69faf-03e1-4762-89b3-f8cf6c29fe24 external_ids:vendor=kube-ovn external_ids:node=eis163 in 10072ms
W0727 15:35:47.881749       1 ovn-sbctl.go:49] ovn-sbctl command took too long: ovn-sbctl --timeout=60 --db=tcp:[10.251.142.69]:6642,tcp:[10.251.142.152]:6642,tcp:[10.251.142.153]:6642 set chassis d0419e34-fc44-4c3c-a166-60786d1b3864 external_ids:vendor=kube-ovn external_ids:node=eis178 in 3711ms
W0727 15:35:53.018725       1 ovn-sbctl.go:49] ovn-sbctl command took too long: ovn-sbctl --timeout=60 --db=tcp:[10.251.142.69]:6642,tcp:[10.251.142.152]:6642,tcp:[10.251.142.153]:6642 set chassis 09873cba-f557-4aa5-a7f5-1fcf35823b35 external_ids:vendor=kube-ovn external_ids:node=eis171 in 4588ms
W0727 15:35:58.541524       1 ovn-sbctl.go:49] ovn-sbctl command took too long: ovn-sbctl --timeout=60 --db=tcp:[10.251.142.69]:6642,tcp:[10.251.142.152]:6642,tcp:[10.251.142.153]:6642 --format=csv --no-heading --data=bare --columns=name find chassis name=8f1befd9-9037-4dcc-9a05-7710327ec903 in 1802ms
W0727 15:36:06.781538       1 ovn-sbctl.go:49] ovn-sbctl command took too long: ovn-sbctl --timeout=60 --db=tcp:[10.251.142.69]:6642,tcp:[10.251.142.152]:6642,tcp:[10.251.142.153]:6642 set chassis 8f1befd9-9037-4dcc-9a05-7710327ec903 external_ids:vendor=kube-ovn external_ids:node=eis164 in 8239ms
W0727 15:36:12.412814       1 ovn-sbctl.go:46] ovn-sbctl command error: ovn-sbctl --timeout=60 --db=tcp:[10.251.142.69]:6642,tcp:[10.251.142.152]:6642,tcp:[10.251.142.153]:6642 set chassis afa34d1c-97ed-4ca9-be68-73f893d0dbb8 external_ids:vendor=kube-ovn external_ids:node=eis170 in 5416ms
E0727 15:36:12.412962       1 init.go:864] failed to set chassis nodeTag: failed to set chassis external_ids, ovn-sbctl: tcp:[10.251.142.69]:6642,tcp:[10.251.142.152]:6642,tcp:[10.251.142.153]:6642: database connection failed ()
, "exit status 1"
E0727 15:36:12.413008       1 klog.go:10] "failed to initialize node chassis" err=<
    failed to set chassis external_ids, ovn-sbctl: tcp:[10.251.142.69]:6642,tcp:[10.251.142.152]:6642,tcp:[10.251.142.153]:6642: database connection failed ()
    , "exit status 1"
 >

```
# Steps to Reproduce the Problem

1.
1.
1.

# Additional Info

- Kubernetes version:

  **Output of `kubectl version`:**

  ```
  (paste your output here)
  ```

- kube-ovn version:
  
  ```
  (paste your output here)
  ```

- operation-system/kernel version:
  
  **Output of `awk -F '=' '/PRETTY_NAME/ { print $2 }' /etc/os-release`:**
  **Output of `uname -r`:**
  ```
  (paste your output here)
  ```

### PR DESCRIPTION
**Pull request by bobz965**
- [ ] Make sure you have followed [Kube-OVN Code Style](https://github.com/kubeovn/kube-ovn/blob/master/CODE_STYLE.md).

### What type of this PR
Examples of user facing changes:
- Bug fixes

### Which issue(s) this PR fixes:
Fixes #3084 

### <samp>🤖 Generated by Copilot at bad86a3</samp>

Improve OVN controller performance by skipping chassis updates for nodes with matching ids. Add a chassis id check in `pkg/controller/init.go`.

### <samp>🤖 Generated by Copilot at bad86a3</samp>

> _`chassis id` check_
> _skips nodes with matching names_
> _snowflakes in OVN_

### <samp>🤖 Generated by Copilot at bad86a3</samp>

*  Add a check to skip nodes with matching chassis id and name ([link](https://github.com/kubeovn/kube-ovn/pull/3083/files?diff=unified&w=0#diff-92aa62435bc2b1b2be34f478f650597068bcf35b8d5975d119f91cf78ac54988R856-R863)) to avoid unnecessary updates to the OVN database

### CODE REVIEW

File: pkg/controller/controller.go [hunk 1]
```diff
@@ -782,6 +782,10 @@ func (c *Controller) Run(ctx context.Context) {
 		util.LogFatalAndExit(err, "failed to initialize ovn resources")
 	}
 
+	if err := c.initNodeChassis(); err != nil {
+		util.LogFatalAndExit(err, "failed to initialize node chassis")
+	}
```
Comment (zhangzujian) [inline pkg/controller/controller.go:787]: Why is this moved?

Comment (bobz965) [reply pkg/controller/controller.go:787]: ipam init 耗时比较久，当出现 sb 不可写（set）操作，就会crash，controller就会重建，ipam又会重新刷，所以将node chassis的set 放在前面，且不再重复set sb chassis操作

File: pkg/controller/init.go [hunk 2]
```diff
@@ -853,6 +853,14 @@ func (c *Controller) initNodeChassis() error {
...
 		chassisName := node.Annotations[util.ChassisAnnotation]
 		if chassisName != "" {
+			existChasisId, err := c.ovnLegacyClient.GetChassis(node.Name)
+			if err != nil {
+				klog.Errorf("failed to get chassis id: %v", err)
+				return err
+			}
+			if existChasisId == chassisName {
+				continue
+			}
```
Comment (zhangzujian) [inline pkg/controller/init.go:863]: `failed to set chassis external_ids, ovn-sbctl: tcp:[10.251.142.69]:6642,tcp:[10.251.142.152]:6642,tcp:[10.251.142.153]:6642: database connection failed ()`
SB set 失败不是 sb 连接失败导致的吗？为什么要修改这里的逻辑？

Comment (bobz965) [reply pkg/controller/init.go:863]: if exist {
				err = c.ovnLegacyClient.InitChassisNodeTag(chassisName, node.Name)
870 行，如果chassis 存在就回去set external id，这个操作几乎都是重复的操作，如果可以get到，应该就不需要重复set了

**Review by zhangzujian — APPROVED** [review]

**Pull request merged**